### PR TITLE
location cleanup to make people more aware of their sandbox placename

### PR DESCRIPTION
**Pull request by howard-stearns**
DNM pending merge/deploy of https://github.com/highfidelity/data-web/pull/876, which is also required for testing.

- Make address bar's "your location", and javacript location.href match existing "Navigate > Copy Address to Clipboard" behavior. Specifically, if you connect to a domain via an ip address or "localhost", or click the home button or use location.goToLocalSandbox(), we resolve to the temp or permanent placename.
- Javascript location.placename() and AddressManger::getPlaceName() also do this resolution. The former is used by the address bar, above, and the latter is used in checking in to the domain server, so that information will now be correct.
- Note that:
 - The application title bar will continue to report 'localhost' rather than the resolved name, in case people are relying on that. We might change this in the future once people are used to seeing their sandbox placename.
 - location.hostname does not change. It currently reports the original ip address or the string 'localhost', without resolving. However, it does resolve a domain-id to the default placename.

### CODE REVIEW

**Comment by howard-stearns** [comment]
test plan:
- In Sandbox Settings, "Disconnect High Fidelity Account" and restart, so that you have a temp name. (You will be able to reconnect later.) Make a note of you XXXXX
 - Go home by clicking the home icon in the GoTo address bar. "Navigate > Copy Address to Clipboard" (which requires "Settings > Advanced Menus" to be enabled), the GoTo address bar, location.href, and location.placename should report your temp name. (Note that the address bar does not list orientation.) The application title bar and location.hostname should say "localhost" (as it does without this PR).
 - Go to a different domain and repeat, making sure the above work as spected.
 - Do location.goToLocalSandbox() in the javascript console and confirm that the results are the same as above.
 - While still in home, take snapshot and confirm that it does not provide a "Share in Feed" button (because we don't currently share from temp domains).
 - In the GoTo address bar, enter your ip address. Confirm that the results are similar to above (with the ip address replacing the string "localhost"). (Depending on your network, you may need to specify a specific port in Sandbox Settings Advanced Local UDP Port, and specify that the goto location as ip-address:port. In any case, the port will not appear in location.hostname or the title bar.)
- In Sandbox Settings, reconnect your account High Fidelity, and repeat the above if you have a permanent place name. Everything should be as above (using this place name), except that now when you take a snapshot, there should be a blue button to "Share in Feed".

**Comment by hifi-gustavo** [comment]
Build console output for commit: da56457c4a181047f3a752c2a0566dec0270ace3 available at: https://gist.github.com/9cb86ed86ca106942054ba8f93b691e8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-da56457.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-da56457.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f1806eb64917687924aceed7165f42b5500a32ac available at: https://gist.github.com/9cb86ed86ca106942054ba8f93b691e8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-f1806eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-f1806eb.dmg

File: scripts/system/snapshot.js [hunk 1]
```diff
@@ -144,7 +164,7 @@ function resetButtons(pathStillSnapshot, pathAnimatedSnapshot, notify) {
         { localPath: pathAnimatedSnapshot },
         { localPath: pathStillSnapshot },
         {
-            canShare: !!location.placename,
+            canShare: !!isDomainOpen(location.domainId),
```
Comment (davidkelly) [inline scripts/system/snapshot.js]: I'm wondering -- does this now share domains that were not shareable before?  It looks at first glance like a domain with no place name.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 066a5181421b1c91d8211b6210f99eaf337ae0ec available at: https://gist.github.com/9cb86ed86ca106942054ba8f93b691e8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-066a518.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-066a518.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2beed7e70b66ded39acc8107741f2cd4d8ccfcd6 available at: https://gist.github.com/9cb86ed86ca106942054ba8f93b691e8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-2beed7e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9096-2beed7e.dmg

**Pull request merged**